File: hivemind.client/src/Types/Lineup.ts
export interface LineupItem {
    lineupItemId: number;
    name: string;
    index: number;
    type: string;
    queries: QueryLineupItem[];
}

export interface QueryLineupItem {
    queryLineupItemId: number;
    queryId: number;
    lineupItemId: number;
    queryType: number;
    index: number;
    padTo: number;
    playCount: number;
    playDuration: number;
    playoutType: number;
}

export interface Lineup {
    lineupId: number;
    lineupName: string;
    channelId: number | null;
    startTime: string;
    lineupItems: LineupItem[];
}

File: HiveMind.Server/Entities/Lineup.cs
namespace HiveMind.Server.Entities;

public class Lineup
{
    public int LineupId { get; set; }
    public string LineupName { get; set; } = string.Empty;
    public TimeOnly StartTime { get; set; }
    public ICollection<LineupItem>? LineupItems { get; set; }
}
